connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -83,8 +86,8 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[ExecutePlanResp
 
       try {
         v.getPlan.getOpTypeCase match {
-          case proto.Plan.OpTypeCase.COMMAND => handleCommand(session, v)
-          case proto.Plan.OpTypeCase.ROOT => handlePlan(session, v)
+          case proto.Plan.OpTypeCase.COMMAND => handleCommand(session, v, sessionHolder.events)
+          case proto.Plan.OpTypeCase.ROOT => handlePlan(session, v, sessionHolder.events)
```

Comment:
I would be in favor of just passing SessionHolder / ExecutePlanHolder everywhere, instead of having to modify these parameters all the time.

Reply:
Gone ahead and done this refactoring.